### PR TITLE
Update SCITT Specs & References

### PR DESCRIPTION
**Pull request by SteveLasker**

### CODE REVIEW

File: scitt-specs.md [hunk 1]
```diff
@@ -17,29 +17,30 @@ The following specifications are part of, or related to the IETF SCITT working g
 - SCITT Reference APIs (SCRAPI)
   - [IETF datatracker](https://datatracker.ietf.org/doc/draft-ietf-scitt-scrapi/){:target="_blank"}
   - [GitHub Repo](https://github.com/ietf-wg-scitt/draft-ietf-scitt-scrapi){:target="_blank"}
+- SCITT COSE Hash Envelope
+  - [GitHub Repo](https://github.com/OR13/draft-steele-cose-hash-envelope){:target="_blank"}
```
Comment (OR13) [inline scitt-specs.md:21]:
```suggestion
  - [GitHub Repo](https://github.com/cose-wg/draft-ietf-cose-hash-envelope){:target="_blank"}
```

Comment (OR13) [reply scitt-specs.md:21]: unclear why some of these are data tracker and others are github repos... data tracker links to github:

https://datatracker.ietf.org/doc/draft-ietf-cose-hash-envelope/

File: scitt-specs.md [hunk 2]
```diff
@@ -17,29 +17,30 @@ The following specifications are part of, or related to the IETF SCITT working g
...
+- COSE Receipts
+  - [IETF datatracker](https://datatracker.ietf.org/doc/draft-birkholz-scitt-receipts/){:target="_blank"}
```
Comment (OR13) [inline scitt-specs.md:23]:
```suggestion
  - [IETF datatracker](https://datatracker.ietf.org/doc/draft-ietf-cose-merkle-tree-proofs/){:target="_blank"}
```

File: scitt-specs.md [hunk 3]
```diff
@@ -17,29 +17,30 @@ The following specifications are part of, or related to the IETF SCITT working g
...
+  - [GitHub Repo](https://github.com/ietf-scitt/draft-birkholz-scitt-receipts){:target="_blank"}
```
Comment (OR13) [inline scitt-specs.md:24]:
```suggestion
  - [GitHub Repo](https://github.com/cose-wg/draft-ietf-cose-merkle-tree-proofs){:target="_blank"}
```

**Review by OR13 — DISMISSED** [review]
see suggestions

**Review by aj-stein — APPROVED** [review]

**Pull request merged**